code/control/DNDeploymentDispatcher.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+use Finite\StateMachine\StateMachine;
+
+class DNDeploymentDispatcher extends Dispatcher {
+
+	const ALLOW_DNDEPLOYMENT_APPROVE = 'ALLOW_DNDEPLOYMENT_APPROVE';
+	const ALLOW_DNDEPLOYMENT_QUEUE = 'ALLOW_DNDEPLOYMENT_QUEUE';
+
+	private static $allowed_actions = [
+		'apply',
+		'canApply',
+	];
+
+	private static $url_handlers = [
+		'$Id/apply/$State' => 'apply',
+		'$Id/can/$State' => 'canApply',
+	];
+
+	public function getModel($name) {
+		$id = $this->request->param('Id');
+		$deployment = DNDeployment::get()->byId($id);
+		return [
+			'Status' => $deployment->Status
```

Comment:
$deployment->State? the $deployment->getStatus() method is just there for legacy compat?